modules/tinymce/src/core/main/ts/delete/TableDeleteUtils.ts

```diff
@@ -20,15 +20,29 @@ const isRootFromElement = (root: SugarElement<Node>): IsRootFn =>
 const getTableCells = (table: SugarElement<HTMLTableElement>): SugarElement<HTMLTableCellElement>[] =>
   SelectorFilter.descendants<HTMLTableCellElement>(table, 'td,th');
 
+const areSameTable = (table1: Optional<SugarElement<HTMLTableElement>>, table2: Optional<SugarElement<HTMLTableElement>>): boolean =>
+  Optionals.lift2(table1, table2, Compare.eq).getOr(false);
```

Comment:
You're having to do a lot of dancing around the logic due to the optional values. Please adjust `getTableDetailsFromRange` to make a new branch of logic that only runs if both start and end table are `some`.

This may also allow you to switch from `let` back to `const`, I'm not sure - perhaps make a complete separate branch for this case and have two `return` blocks.

Reply:
@TheSpyder I refactored it making a function that manage the quirks for details and making the specific case for this issue, let me know what do you think about it

Reply:
It's certainly improved, but I encourage you to try for something better. I was hinting at something important - optional values are getting in your way 🙂 

Your logic requires start and end to be `some`, so start by unwrapping them. This would be along the lines of `Optionals.lift2(maybeStart, maybeEnd, (start, end) => { // logic here }).getOr(details)`. The `lift2` function is for much more than just comparison.

That will greatly simplify the code and remove the need for this `areSameTable` function completely; you'll be able to just use `Compare.eq` in your code.